framework/Samplers/AdaptiveMonteCarlo.py

```diff
@@ -0,0 +1,226 @@
+      statEr, ste = metric.split('_')
+      if statEr in cls.statScVals:
+        statErSpecification = InputData.parameterInputFactory(statEr, contentType=InputTypes.StringListType)
+        statErSpecification.addParam("prefix", InputTypes.StringType)
+        statErSpecification.addParam("tol", InputTypes.FloatType)
+        convergenceInput.addSub(statErSpecification)
+    inputSpecification.addSub(convergenceInput)
+    targetEvaluationInput = InputData.parameterInputFactory("TargetEvaluation", contentType=InputTypes.StringType)
+    targetEvaluationInput.addParam("type", InputTypes.StringType)
+    targetEvaluationInput.addParam("class", InputTypes.StringType)
+    inputSpecification.addSub(targetEvaluationInput)
+    inputSpecification.addSub(InputData.parameterInputFactory("initialSeed", contentType=InputTypes.IntegerType))
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Default Constructor that will initialize member variables with reasonable
+      defaults or empty lists/dictionaries where applicable.
+      @ In, None
+      @ Out, None
+    """
+    AdaptiveSampler.__init__(self)
+    MonteCarlo.__init__(self)
+    self.persistence         = 5                # this is the number of times the error needs to fell below the tolerance before considering the sim converged
+    self.persistenceCounter  = 0                # Counter for the persistence
+    self.forceIteration      = False            # flag control if at least a self.limit number of iteration should be done
+    self.solutionExport      = None             # data used to export the solution (it could also not be present)
+    self.tolerance           = {}               # dictionary stores the tolerance for each variables
+    self.converged           = False            # flag convergence
+    self.basicStatPP         = None             # post-processor to compute the basic statistics
+    self.converged           = False            # flag that is set to True when the sampler converged
+    self.printTag            = 'SAMPLER ADAPTIVE MC'
+    self.addAssemblerObject('TargetEvaluation','n')
+
+  def localInputAndChecks(self,xmlNode, paramInput):
+    """
+      Class specific xml inputs will be read here and checked for validity.
+      @ In, xmlNode, xml.etree.ElementTree.Element, The xml element node that will be checked against the available options specific to this Sampler.
+      @ In, paramInput, InputData.ParameterInput, the parsed parameters
+      @ Out, None
+    """
+    self.toDo = {}
+    for child in paramInput.subparts:
+      if child.getName() == "Convergence":
+        for grandchild in child.subparts:
+          tag = grandchild.getName()
+          if tag == "limit":
+            self.limit = grandchild.value
+            if self.limit is None:
+              self.raiseAnError(IOError,self,'Adaptive Monte Carlo sampler '+self.name+' needs the limit block (number of samples) in the Convergence block')
```

Comment:
This should not be needed since the InputParameter above checks the required input for you.

Reply:
If it is not reached the convergence it will run forever? It does not check the InputParameter in the Monte Carlo in initSampler actually.

Reply:
limit is a required parameter... If not inputted the InputData should error out before we reach this line.